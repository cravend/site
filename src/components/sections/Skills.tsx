import { useTranslations } from "next-intl";

const Skills = () => {
  const t = useTranslations("pages.Index");
  return (
    <>
      <h2>{t.rich("skills.header")}</h2>
      <ul>
        <li>{t.rich("skills.languages_spoken")}</li>
        <li>{t.rich("skills.languages_tech")}</li>
        <li>{t.rich("skills.tools")}</li>
        <li>
          <strong>{t.rich("skills.certifications.header")}</strong>
          <ul>
            <li>{t.rich("skills.certifications.content.0")}</li>
            <li>{t.rich("skills.certifications.content.1")}</li>
            <li>{t.rich("skills.certifications.content.2")}</li>
          </ul>
        </li>
      </ul>
    </>
  );
};

export default Skills;
