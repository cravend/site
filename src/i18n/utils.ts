import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
