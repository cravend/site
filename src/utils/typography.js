import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
