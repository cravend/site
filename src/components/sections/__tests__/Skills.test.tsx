import { render } from "../../../tests/utils";
import Skills from "../Skills";

describe("<Skills />", () => {
  it("renders", () => {
    expect.assertions(1);
    const { container } = render(<Skills />);
    expect(container).toMatchSnapshot();
  });
});
