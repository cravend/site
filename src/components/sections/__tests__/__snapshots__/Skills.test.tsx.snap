// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skills /> renders 1`] = `
<div>
  <h2>
    Skills
  </h2>
  <ul>
    <li>
      <strong>
        Languages (spoken):
      </strong>
       
      <a
        class="primary"
        href="/"
      >
        English
      </a>
      , 
      <a
        class="primary"
        href="/"
      >
        French
      </a>
      , American Sign Language (ASL), 
      <a
        class="primary"
        href="/"
      >
        Arabic
      </a>
    </li>
    <li>
      <strong>
        Languages (tech):
      </strong>
       React, GraphQL, Java, HTML, CSS, JavaScript, Visual Basic .NET, C++, PHP, SQL, Python, R
    </li>
    <li>
      <strong>
        Tools:
      </strong>
       Next, Git, GitHub, GitLab, Gatsby, cPanel, WHM, Flask, Django, Linux, DigitalOcean, AWS, Netlify, WordPress
    </li>
    <li>
      <strong>
        Certifications
      </strong>
      <ul>
        <li>
          The Fundamentals of Digital Marketing, Google — 
          <em>
            Certified April 2020
          </em>
        </li>
        <li>
          Microsoft Office Specialist (MOS) Master 2016 — 
          <em>
            Certified fall 2018
          </em>
        </li>
        <li>
          MOS Word 2016 Expert, MOS Excel 2016 Expert — 
          <em>
            Certified fall 2018
          </em>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
